### PR TITLE
Event model

### LINKED ISSUE #6: Event model

### PR DESCRIPTION
**Pull request by LatidoReMe**
close #6 

Notes: I plan to squash all migration files @ the end, since I might need to make more migrations with these fixes later on.

### CODE REVIEW

File: sukutomo/models.py [hunk 1]
```diff
@@ -236,3 +238,99 @@ def astrological_sign_image_url(self): return staticImageURL(self.i_astrological
...
+    jp_banner = models.ImageField(string_concat(t['Japanese'], ' ', _('version'), '-',_('Banner')), upload_to=uploadItem('e'), null=True)
+    jp_start_date = models.DateTimeField(string_concat(t['Japanese'], ' ', _('version'), ' - ', _('Beginning')), null=True)
+    jp_end_date = models.DateTimeField(string_concat(t['Japanese'], ' ', _('version'), ' - ',_('End')), null=True)
+
+    ww_title = models.CharField(string_concat(_('Title'),  ' (', _('Worldwide'), ')'), max_length=100, null=True)
+    ww_banner = models.ImageField(string_concat(_('Worldwide'), ' ', _('version'), '-',_('Banner')), upload_to=uploadItem('e'), null=True)
+    ww_start_date = models.DateTimeField(string_concat(_('Worldwide'), ' ', _('version'), ' - ', _('Beginning')), null=True)
+    ww_end_date = models.DateTimeField(string_concat(_('Worldwide'), ' ', _('version'), ' - ',_('End')), null=True)
+
+    tw_title = models.CharField(string_concat(_('Title'), ' (', _('Taiwanese'), ')'), max_length=100, null=True)
```
Comment (db0company) [inline sukutomo/models.py]: Didn't tw merge with en? 🤔

File: sukutomo/magicollections.py [hunk 2]
```diff
@@ -84,6 +84,7 @@ class IdolCollection(MagiCollection):
     multipart = True
     form_class = forms.IdolForm
     reportable = False
+    blockable = False
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/magicollections.py]: ✅

File: sukutomo/migrations/0026_auto_20180427_2014.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='event',
+            name='c_versions',
+            field=models.TextField(default=b'"JP"', null=True, verbose_name='Server availability', blank=True),
+            preserve_default=True,
+        ),
+    ]
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/migrations/0026_auto_20180427_2014.py]: ✅

File: sukutomo/models.py [hunk 4]
```diff
@@ -236,3 +238,99 @@ def astrological_sign_image_url(self): return staticImageURL(self.i_astrological
...
+class Event(MagiModel):
+    collection_name = 'event'
+    owner = models.ForeignKey(User, related_name='added_events')
+
+    title = models.CharField(_('Title'), max_length=100)
+    TITLES_CHOICES = ALT_LANGUAGES_EVENT_TITLES
+    d_titles = models.TextField(null=True)
+
+    banner = models.ImageField(_('Banner'), null=True)
+    start_date = models.DateTimeField(_('Beginning'), null=True)
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

File: sukutomo/forms.py [hunk 5]
```diff
@@ -46,4 +46,25 @@ class Meta:
         model = models.Idol
         fields = ('search', 'i_attribute', 'i_unit', 'i_subunit', 'i_school', 'i_year', 'i_astrological_sign', 'i_blood')
 
-            
+class EventForm(AutoForm):
+    start_date = forms.forms.DateField(label=_('Beginning'), required=False)
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/forms.py]: ✅

File: sukutomo/models.py (hunk 1 above)
Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

File: sukutomo/models.py [hunk 6]
```diff
@@ -102,6 +103,7 @@ def update_cache_leaderboards(self):
 
 LANGUAGES_NEED_OWN_NAME = [ l for l in django_settings.LANGUAGES if l[0] in ['ru', 'zh-hans', 'zh-hant', 'kr'] ]
 ALL_ALT_LANGUAGES = [ l for l in django_settings.LANGUAGES if l[0] != 'en' ]
+ALT_LANGUAGES_EVENT_TITLES= [ l for l in django_settings.LANGUAGES if l[0] not in ['en', 'ja', 'kr', 'zh-hant', 'zh-hans'] ]
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

File: sukutomo/models.py [hunk 7]
```diff
@@ -236,3 +238,99 @@ def astrological_sign_image_url(self): return staticImageURL(self.i_astrological
...
+    TITLES_CHOICES = ALT_LANGUAGES_EVENT_TITLES
+    d_titles = models.TextField(null=True)
+
+    banner = models.ImageField(_('Banner'), null=True)
+    start_date = models.DateTimeField(_('Beginning'), null=True)
+    end_date = models.DateTimeField(_('End'), null=True)
+
+    @property
+    def t_name(self):
+        if get_language() == 'ja':
```
[the remark this answers is not in the document]
Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

File: sukutomo/models.py [hunk 8]
```diff
@@ -247,26 +246,11 @@ class Event(MagiModel):
     owner = models.ForeignKey(User, related_name='added_events')
 
     title = models.CharField(_('Title'), max_length=100)
-    TITLES_CHOICES = ALT_LANGUAGES_EVENT_TITLES
+    TITLES_CHOICES = ALL_ALT_LANGUAGES
     d_titles = models.TextField(null=True)
-
+    
+    jp_title = models.CharField(string_concat(_('Title'), ' (', t['Japanese'], ')'), max_length=100, null=True)
```
Comment (db0company) [inline sukutomo/models.py]: Why this one separate? You can also have it in the d_titles!

Comment (LatidoReMe) [reply sukutomo/models.py]: I thought you'd want to store the japanese name seperate for the event like you did on BanPa, but if you're ok without it, I'll just use d_titles instead

Comment (LatidoReMe) [reply sukutomo/models.py]: ✅

**Pull request merged**